apps/pay/app/api/lnurl-proxy/route.ts

```diff
@@ -0,0 +1,51 @@
+import { NextRequest, NextResponse } from "next/server"
+import { getParams } from "js-lnurl"
+
+export async function POST(request: NextRequest) {
+  try {
+    const { lnurl, paymentRequest } = await request.json()
+
+    if (!lnurl) {
+      return NextResponse.json(
+        { error: "Missing lnurl parameter" },
+        { status: 400 }
+      )
+    }
+
+    // If we have both lnurl and paymentRequest, we're doing a direct payment
+    if (lnurl && paymentRequest) {
```

Comment:
if we need both then it is cleaner to add paymentRequest validation and remove this if `if (lnurl && paymentRequest)`
```js
    if (!paymentRequest) {
      return NextResponse.json(
        { error: "Missing paymentRequest parameter" },
        { status: 400 }
      )
    }
```